Artifacts/MsSql/Structure/Security/0040-Tables-MultipleAuthStrats.sql

```diff
@@ -0,0 +1,131 @@
+ALTER TABLE [dbo].[ClaimSetResourceClaimActions]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActions_dbo.ResourceClaims_ResourceClaimId] FOREIGN KEY([ResourceClaimId])
+REFERENCES [dbo].[ResourceClaims] ([ResourceClaimId])
+GO
+
+---------------------------- [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] --------------------------------------------------------
+
+CREATE TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides](
+	[ClaimSetResourceClaimActionAuthorizationStrategyOverrideId] [int] IDENTITY(1,1) NOT NULL,
+	[ClaimSetResourceClaimActionId] [int] NOT NULL,
+	[AuthorizationStrategyId] [int] NOT NULL,
+ CONSTRAINT [PK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides] PRIMARY KEY CLUSTERED 
+(
+	[ClaimSetResourceClaimActionAuthorizationStrategyOverrideId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+-- NONCLUSTERED INDEX for [ClaimSetResourceClaimActionId],[AuthorizationStrategyId] columns 
+CREATE NONCLUSTERED INDEX [IX_ClaimSetResourceClaimActionId_AuthorizationStrategyId]
+    ON [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]([ClaimSetResourceClaimActionId],[AuthorizationStrategyId] ASC);
+GO
+
+-- FOREIGN KEY for [AuthorizationStrategyId]  column
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.AuthorizationStrategies_AuthorizationStrategyId] FOREIGN KEY([AuthorizationStrategyId])
+REFERENCES [dbo].[AuthorizationStrategies] ([AuthorizationStrategyId])
+ON DELETE CASCADE
+GO
+
+-- FOREIGN KEY for [ClaimSetResourceClaimActionId]  column
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.ClaimSetResourceClaimActions_ClaimSetResourceClaimActionId] FOREIGN KEY([ClaimSetResourceClaimActionId])
+REFERENCES [dbo].[ClaimSetResourceClaimActions] ([ClaimSetResourceClaimActionId])
+GO
+
+------------------------------ [dbo].[ResourceClaimActionAuthorizationStrategies] -------------------------------------------------
+
+CREATE TABLE [dbo].[ResourceClaimActionAuthorizationStrategies](
+	[ResourceClaimActionAuthorizationStrategyId] [int] IDENTITY(1,1) NOT NULL,
+	[ResourceClaimActionId] [int] NOT NULL,
+	[AuthorizationStrategyId] [int] NOT NULL,
+
+ CONSTRAINT [PK_dbo.ResourceClaimActionAuthorizationStrategies] PRIMARY KEY CLUSTERED 
+(
+	[ResourceClaimActionAuthorizationStrategyId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+-- NONCLUSTERED INDEX for [AuthorizationStrategyId],[ResourceClaimActionId] columns 
+CREATE NONCLUSTERED INDEX [IX_AuthorizationStrategyId_ResourceClaimActionId]
```

Comment:
This needs to be a _unique_ constraint/index, and the `ResourceClaimActionId` should be first, followed by `AuthorizationStrategyId`.

Reply:
I created CREATE UNIQUE INDEX here